src/juju/jimm-facade.ts

```diff
@@ -3,10 +3,32 @@ import { autoBind } from "@canonical/jujulib/dist/api/utils";
 
 import type { Controller } from "store/juju/types";
 
+// As typed in JIMM:
+// https://github.com/canonical/jimm/blob/1da76ed2d8dba741f5880f32c073f85f7d518904/api/params/params.go#L344
+export type CrossModelQueryRequest = {
+  type: string;
+  query: string;
+};
+
+// As typed in JIMM:
+// https://github.com/canonical/jimm/blob/1da76ed2d8dba741f5880f32c073f85f7d518904/api/params/params.go#L353
+export type CrossModelQueryResponse = {
+  results: Record<string, unknown[]>;
+  errors: Record<string, string[]>;
+};
+
+// In the case an invalid type and/or query is passed, the response might
+// sometimes be a string.
+export type CrossModelQueryFullResponse = CrossModelQueryResponse | string;
+
+export const isCrossModelQueryResponse = (
+  response: CrossModelQueryFullResponse
+): response is CrossModelQueryResponse => typeof response === "object";
```

Comment:
Nice work adding this. On little issue is that `typeof null === "object"` as well so this should probably also check that the value is truthy.

```suggestion
): response is CrossModelQueryResponse => response && typeof response === "object";
```

Reply:
Ah yes of course, I think it's fine how it is then.